### PR TITLE
WIP: Add island and ship page

### PR DESCRIPTION
**Pull request by outdoteth**
Screenshot 2021-09-10 at 22 33 57
Screenshot 2021-09-10 at 22 34 37

### CODE REVIEW

File: components/ResourceBalance.tsx [hunk 1]
```diff
@@ -26,25 +31,35 @@ const resourceToContract: {
...
-  const {
-    data
-  } = useSWR(['balanceOf', contractAddress, account],
-    (_, address, account) => BaseErc20Factory.connect(address, defaultProvider).balanceOf(account)
-  )
-
-  if (data?.lte('0')) {
-    return <div />
+  if (data?.lte("0")) {
+    return <div />;
```
Comment (ethandaya) [inline components/ResourceBalance.tsx]: nit: can just return null

File: compositions/ManageIsland.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  id: string;
+}
+
+export const ManageIsland: React.FC<ManageIslandProps> = ({ id }) => {
+  const { Islands, isReadOnly } = useContext(ContractContext);
+  const {
+    handleTx: handleHarvest,
+    txStatus: harvestStatus,
+    txInProgress: harvIP,
+  } = useContractTransaction(1);
```
Comment (ethandaya) [inline compositions/ManageIsland.tsx]: the hook will default to waiting 1

File: compositions/ManageShip.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    txInProgress: harvIP,
+  } = useContractTransaction(1);
+
+  const harvest = useCallback(async () => {
+    try {
+      if (!Ships || isReadOnly) {
+        throw new Error("Contract is not authorised");
+      }
+
+      await handleHarvest(Ships.harvest(id));
```
Comment (ethandaya) [inline compositions/ManageShip.tsx:23]: do we need an approval between V2 and ships, or can it use public functions

File: pages/island/[id].tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    </Fragment>
+  );
+};
+
+export const getStaticProps: GetServerSideProps<Props> = async ({ params }) => {
+  if (!params?.id || Array.isArray(params.id)) {
+    return { notFound: true };
+  }
+
+  const id = params.id as string;
```
Comment (ethandaya) [inline pages/island/[id].tsx]: if you pass a URLParams type as the second type for the "GetSaticProps" type it'll have the correct types

File: pages/island/[id].tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+              ))}
+            </div>
+          )}
+        </NFTFullPage>
+      </main>
+    </Fragment>
+  );
+};
+
+export const getStaticProps: GetServerSideProps<Props> = async ({ params }) => {
```
Comment (ethandaya) [inline pages/island/[id].tsx]: should be GetSaticProps not GetServerSde

**Review by ethandaya — COMMENTED** [review]
this all looks great sans some nits

File: compositions/ManageShip.tsx (hunk 3 above)
Comment (outdoteth) [reply compositions/ManageShip.tsx:23]: It can use public functions

Comment (outdoteth) [reply compositions/ManageShip.tsx:23]: We need to approve the ships contract and islands contract as minters on the wood, iron and silver token contracts though I think that is separate.

**Pull request merged**